design/disaster-recovery-snapshots.md

```diff
@@ -0,0 +1,128 @@
+# Optionally enable disaster recovery snapshots
+
+Currently KOTS supports application level snapshots, which can be used to restore a previously deployed application version along with its data.
+However in case of a disaster, only application can be restored to a new cluster and Admin Console can never be used to manage it again.
+This document proposes a way to implement disaster recovery snapshots using Velero.
+
+## Goals
+
+1. Use Velero to create Admin Console snapshot.
+1. Make CLI based disaster recovery possible.
+1. Propose a way to enable UI based recovery process.
+
+## Non Goals
+
+1. Improving application deployment process to avoid the `field is immutable` errors.
+1. Performing incremental backups of docker images.
+
+## Background
+
+Velero allows a number of ways to include and exclude resources in backups.
+In current implememntation, backup is created with the `includedNamespaces` key set to include all namespaces where application is deployed.
+All Admin Console resources have label `velero.io/exclude-from-backup: "true"`, which is recognized by Velero to exclude these resources.
+
+
+Disaster recovery snapshots must include Admin Console data, and potentially all resources.
+Due to this requirement, `velero.io/exclude-from-backup` label can no longer be used.
+The `includedNamespaces` filter also cannot be used because Admin Console and application can be deployed into the same namespace, which will cause problems during the restore process.
+
+## High-Level Design
+
+Velero backup spec supports the `labelSelector` field.
+All resources, application and Admin Console, can be labeled for inclusion in snapshots.
+Similarly, restore spec supports the same `labelSelector` field to select specific resources to restore.
+Using different `labelSelector` values allows to restore application and Admin Console separately.
+
+## Detailed Design
+
+### Backup
+
+- All resources (Admin Console and application) will have an additional label:
+    - `kots.io/backup: velero`
```

Comment:
If you're using `kots.io/backup`, do you really need `velero`? Velero is an implementation detail.

e.g. maybe use `kots.io/backup: kotsadm` instead, and then applications can use `kots.io/backup: application` or even just `kots.io/backup: <app-slug>`. 

Assuming the goal is to be able to differentiate admin console content from application content, that should be sufficient.

Does `matchLabels` only work with a single label? Is that why you want to structure the labels this way?

Reply:
Do applications ever create more objects that need to be backed up, that Kots wouldn't be tracking?